Compile error

Hi, I just found your repo and tried it out, but it won't compile
The library does not seem to be detected as such after installing it, so I included the hw_rng.h file in the example manually.

Anything I'm missing here?
<details><summary>Click to expand</summary><P/>

```
Arduino: 1.8.12 (Mac OS X), Board: "Nordic nRF52840DK (PCA10056), 0.3.2 SoftDevice s140 6.1.1, Level 0 (Release)"

/Applications/Arduino.app/Contents/Java/arduino-builder -dump-prefs -logger=machine -hardware /Applications/Arduino.app/Contents/Java/hardware -hardware /Users/player1/Library/Arduino15/packages -hardware /Users/player1/Synced/Documents/Arduino/hardware -tools /Applications/Arduino.app/Contents/Java/tools-builder -tools /Applications/Arduino.app/Contents/Java/hardware/tools/avr -tools /Users/player1/Library/Arduino15/packages -built-in-libraries /Applications/Arduino.app/Contents/Java/libraries -libraries /Users/player1/Synced/Documents/Arduino/libraries -fqbn=adafruit:nrf52:pca10056:softdevice=s140v6,debug=l0 -ide-version=10812 -build-path /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356 -warnings=none -build-cache /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_cache_819538 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.nrfjprog.path=/Users/player1/Library/Arduino15/packages/adafruit/tools/nrfjprog/9.4.0 -prefs=runtime.tools.nrfjprog-9.4.0.path=/Users/player1/Library/Arduino15/packages/adafruit/tools/nrfjprog/9.4.0 -prefs=runtime.tools.arm-none-eabi-gcc.path=/Users/player1/Library/Arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4 -prefs=runtime.tools.arm-none-eabi-gcc-7-2017q4.path=/Users/player1/Library/Arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4 -verbose /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino
/Applications/Arduino.app/Contents/Java/arduino-builder -compile -logger=machine -hardware /Applications/Arduino.app/Contents/Java/hardware -hardware /Users/player1/Library/Arduino15/packages -hardware /Users/player1/Synced/Documents/Arduino/hardware -tools /Applications/Arduino.app/Contents/Java/tools-builder -tools /Applications/Arduino.app/Contents/Java/hardware/tools/avr -tools /Users/player1/Library/Arduino15/packages -built-in-libraries /Applications/Arduino.app/Contents/Java/libraries -libraries /Users/player1/Synced/Documents/Arduino/libraries -fqbn=adafruit:nrf52:pca10056:softdevice=s140v6,debug=l0 -ide-version=10812 -build-path /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356 -warnings=none -build-cache /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_cache_819538 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.nrfjprog.path=/Users/player1/Library/Arduino15/packages/adafruit/tools/nrfjprog/9.4.0 -prefs=runtime.tools.nrfjprog-9.4.0.path=/Users/player1/Library/Arduino15/packages/adafruit/tools/nrfjprog/9.4.0 -prefs=runtime.tools.arm-none-eabi-gcc.path=/Users/player1/Library/Arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4 -prefs=runtime.tools.arm-none-eabi-gcc-7-2017q4.path=/Users/player1/Library/Arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4 -verbose /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino
Using board 'pca10056' from platform in folder: /Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1
Using core 'nRF5' from platform in folder: /Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1
Detecting libraries used...
/Users/player1/Library/Arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/arm-none-eabi-g++ -mcpu=cortex-m4 -mthumb -c -g -w -mfloat-abi=hard -mfpu=fpv4-sp-d16 -u _printf_float -std=gnu++11 -ffunction-sections -fdata-sections -fno-threadsafe-statics -nostdlib --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -w -x c++ -E -CC -DF_CPU=64000000 -DARDUINO=10812 -DARDUINO_NRF52840_PCA10056 -DARDUINO_ARCH_NRF52 "-DARDUINO_BSP_VERSION=\"0.20.1\"" -DNRF52840_XXAA -DUSBCON -DUSE_TINYUSB -DUSB_VID=0x239A -DUSB_PID=0x8029 "-DUSB_MANUFACTURER=\"Nordic\"" "-DUSB_PRODUCT=\"nRF52840 DK\"" -DSOFTDEVICE_PRESENT -DARDUINO_NRF52_ADAFRUIT -DNRF52_SERIES -DLFS_NAME_MAX=64 -Ofast -DCFG_DEBUG=0 -DCFG_LOGGER=1 -DCFG_SYSVIEW=0 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/cmsis/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/hal -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/mdk -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/soc -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/drivers/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/drivers/src -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/softdevice/s140_nrf52_6.1.1_API/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/Source/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/config -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/portable/GCC/nrf52 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/portable/CMSIS/nrf52 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/sysview/SEGGER -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/sysview/Config -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB/Adafruit_TinyUSB_ArduinoCore -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB/Adafruit_TinyUSB_ArduinoCore/tinyusb/src -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/variants/pca10056 /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng_test01.ino.cpp -o /dev/null
Generating function prototypes...
/Users/player1/Library/Arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/arm-none-eabi-g++ -mcpu=cortex-m4 -mthumb -c -g -w -mfloat-abi=hard -mfpu=fpv4-sp-d16 -u _printf_float -std=gnu++11 -ffunction-sections -fdata-sections -fno-threadsafe-statics -nostdlib --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -w -x c++ -E -CC -DF_CPU=64000000 -DARDUINO=10812 -DARDUINO_NRF52840_PCA10056 -DARDUINO_ARCH_NRF52 "-DARDUINO_BSP_VERSION=\"0.20.1\"" -DNRF52840_XXAA -DUSBCON -DUSE_TINYUSB -DUSB_VID=0x239A -DUSB_PID=0x8029 "-DUSB_MANUFACTURER=\"Nordic\"" "-DUSB_PRODUCT=\"nRF52840 DK\"" -DSOFTDEVICE_PRESENT -DARDUINO_NRF52_ADAFRUIT -DNRF52_SERIES -DLFS_NAME_MAX=64 -Ofast -DCFG_DEBUG=0 -DCFG_LOGGER=1 -DCFG_SYSVIEW=0 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/cmsis/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/hal -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/mdk -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/soc -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/drivers/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/drivers/src -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/softdevice/s140_nrf52_6.1.1_API/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/Source/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/config -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/portable/GCC/nrf52 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/portable/CMSIS/nrf52 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/sysview/SEGGER -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/sysview/Config -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB/Adafruit_TinyUSB_ArduinoCore -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB/Adafruit_TinyUSB_ArduinoCore/tinyusb/src -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/variants/pca10056 /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng_test01.ino.cpp -o /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/preproc/ctags_target_for_gcc_minus_e.cpp
/Applications/Arduino.app/Contents/Java/tools-builder/ctags/5.8-arduino11/ctags -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/preproc/ctags_target_for_gcc_minus_e.cpp
Compiling sketch...
/Users/player1/Library/Arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/arm-none-eabi-g++ -mcpu=cortex-m4 -mthumb -c -g -w -mfloat-abi=hard -mfpu=fpv4-sp-d16 -u _printf_float -std=gnu++11 -ffunction-sections -fdata-sections -fno-threadsafe-statics -nostdlib --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -MMD -DF_CPU=64000000 -DARDUINO=10812 -DARDUINO_NRF52840_PCA10056 -DARDUINO_ARCH_NRF52 "-DARDUINO_BSP_VERSION=\"0.20.1\"" -DNRF52840_XXAA -DUSBCON -DUSE_TINYUSB -DUSB_VID=0x239A -DUSB_PID=0x8029 "-DUSB_MANUFACTURER=\"Nordic\"" "-DUSB_PRODUCT=\"nRF52840 DK\"" -DSOFTDEVICE_PRESENT -DARDUINO_NRF52_ADAFRUIT -DNRF52_SERIES -DLFS_NAME_MAX=64 -Ofast -DCFG_DEBUG=0 -DCFG_LOGGER=1 -DCFG_SYSVIEW=0 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/cmsis/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/hal -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/mdk -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/soc -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/drivers/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/drivers/src -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/softdevice/s140_nrf52_6.1.1_API/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/Source/include -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/config -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/portable/GCC/nrf52 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/freertos/portable/CMSIS/nrf52 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/sysview/SEGGER -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/sysview/Config -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB/Adafruit_TinyUSB_ArduinoCore -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/TinyUSB/Adafruit_TinyUSB_ArduinoCore/tinyusb/src -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5 -I/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/variants/pca10056 /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng_test01.ino.cpp -o /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng_test01.ino.cpp.o
In file included from /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino:1:0:
/var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng.h: In static member function 'static void SimpleHacks::HW_RNG::begin()':
hw_rng.h:75:45: error: too few arguments to function 'void nrf_rng_error_correction_enable(NRF_RNG_Type*)'
             nrf_rng_error_correction_enable();

                                             ^
In file included from /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng.h:30:0,
                 from /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino:1:
/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/hal/nrf_rng.h:256:24: note: declared here
 NRF_STATIC_INLINE void nrf_rng_error_correction_enable(NRF_RNG_Type * p_reg)
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino:1:0:
hw_rng.h:76:66: error: cannot convert 'nrf_rng_short_mask_t' to 'NRF_RNG_Type*' for argument '1' to 'void nrf_rng_shorts_disable(NRF_RNG_Type*, uint32_t)'
             nrf_rng_shorts_disable(NRF_RNG_SHORT_VALRDY_STOP_MASK);

                                                                  ^
hw_rng.h:77:52: error: cannot convert 'nrf_rng_task_t' to 'NRF_RNG_Type*' for argument '1' to 'void nrf_rng_task_trigger(NRF_RNG_Type*, nrf_rng_task_t)'
             nrf_rng_task_trigger(NRF_RNG_TASK_START);

                                                    ^
In file included from /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino:1:0:
/var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng.h: In static member function 'static uint8_t SimpleHacks::HW_RNG::internal_get_byte()':
hw_rng.h:141:21: error: 'nrf_rng_event_get' was not declared in this scope
             while (!nrf_rng_event_get(NRF_RNG_EVENT_VALRDY));

                     ^~~~~~~~~~~~~~~~~
/var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng.h:141:21: note: suggested alternative: 'nrf_rng_event_t'
             while (!nrf_rng_event_get(NRF_RNG_EVENT_VALRDY));

                     ^~~~~~~~~~~~~~~~~
                     nrf_rng_event_t
hw_rng.h:142:50: error: too few arguments to function 'uint8_t nrf_rng_random_value_get(const NRF_RNG_Type*)'
             uint8_t v = nrf_rng_random_value_get();

                                                  ^
In file included from /var/folders/zm/4nmmy__d0n5_m634_w_hm_1w0000gp/T/arduino_build_548356/sketch/hw_rng.h:30:0,
                 from /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino:1:
/Users/player1/Library/Arduino15/packages/adafruit/hardware/nrf52/0.20.1/cores/nRF5/nordic/nrfx/hal/nrf_rng.h:251:27: note: declared here
 NRF_STATIC_INLINE uint8_t nrf_rng_random_value_get(NRF_RNG_Type const * p_reg)
                           ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/player1/Synced/Documents/Arduino/Libraries/hw_rng/examples/hw_rng_test01/hw_rng_test01.ino:1:0:
hw_rng.h:143:53: error: cannot convert 'nrf_rng_event_t' to 'NRF_RNG_Type*' for argument '1' to 'void nrf_rng_event_clear(NRF_RNG_Type*, nrf_rng_event_t)'
             nrf_rng_event_clear(NRF_RNG_EVENT_VALRDY);

                                                     ^
exit status 1
too few arguments to function 'void nrf_rng_error_correction_enable(NRF_RNG_Type*)'
```
</details>

--- a/changelog.md
+++ b/changelog.md
@@ -1,0 +1,21 @@
+# Changelist for HW_RNG arduino library
+
+## v1.1.0
+
+<details><summary>Fix: #2 by supporting both `nrfx v1.7.2` and `nrfx v2.1.0`</summary><P/>
+
+Nordic made significant breaking API changes to the RNG API
+somewhere between `nrfx v1.7.2` (which v1.0.0 of this library
+was written for) and `nrfx v2.1.0`. These breaking API changes
+were **not** listed in the nrfx changelog.
+
+By sheer luck, a new macro was also defined with this version
+bump, allowing this library to support either version of
+the `nrfx` headers, with zero code change by library users.
+</details><HR/>
+
+Thanks to user @BitLox for reporting this issue, and his help testing the fix.
+
+## v1.0.0
+
+First release.  Works with nRF52 series chips when using Adafruit Arduino BSP.

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=nrf_rng
-version=1.0.0
+version=1.1.0
 author=SimpleHacks
 maintainer=SimpleHacks
 sentence=True non-deterministic random numbers for nRF5 series chips.  MIT License.

--- a/src/hw_rng.h
+++ b/src/hw_rng.h
@@ -51,6 +51,25 @@
     #error "Library currenly only known to support NRF52 architecture"
 #endif
 
+// Somewhere between nrfx v1.7.2 (which this library was originally written for) and
+// nrfx v2.1.0, Nordic made significant breaking changes to their API.  This breaking
+// API change is not listed in the nrfx changelog.
+// 
+// It is desirable to hide this breaking API change from users.
+//
+// [HACK] check for existence of NRFX_STATIC_INLINE, which was not defined
+//        in nrfx v1.x, but is defined in nrfx v2.x.
+//        This works even if NRFX_STATIC_INLINE is defined to nothing
+//
+#if defined(NRFX_STATIC_INLINE)
+    #define SIMPLEHACKS_HW_RNG_NRFX_API_VERSION 2
+    // #warn  "Discovered nrfx v2.x.x APIs ... "
+#else
+    #define SIMPLEHACKS_HW_RNG_NRFX_API_VERSION 1
+    // #warn "Discovered nrfx v1.x.x APIs ... "
+#endif
+
+
 namespace SimpleHacks {
 
     // Difficulty: the RNG is used by the SoftDevice.
@@ -72,9 +91,19 @@ namespace SimpleHacks {
         HW_RNG& operator=(const HW_RNG&&) =delete;
     public: // user API
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static void begin() {
+#if SIMPLEHACKS_HW_RNG_NRFX_API_VERSION == 2
+            // Nordic changes API between nrfx v1.7.2 and v2.1.0...
             nrf_rng_error_correction_enable(NRF_RNG);
             nrf_rng_shorts_disable(NRF_RNG, NRF_RNG_SHORT_VALRDY_STOP_MASK);
             nrf_rng_task_trigger(NRF_RNG, NRF_RNG_TASK_START);
+#elif SIMPLEHACKS_HW_RNG_NRFX_API_VERSION == 1
+            // Nordic changes API between nrfx v1.7.2 and v2.1.0...
+            nrf_rng_error_correction_enable();
+            nrf_rng_shorts_disable(NRF_RNG_SHORT_VALRDY_STOP_MASK);
+            nrf_rng_task_trigger(NRF_RNG_TASK_START);
+#else
+    #error "Unknown NRFX API version ... aborting compilation"
+#endif
         }
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static void end() {}
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static uint8_t get_uint8() {
@@ -138,9 +167,19 @@ namespace SimpleHacks {
 
     private: // to support one-time init...
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static uint8_t internal_get_byte() {
+#if SIMPLEHACKS_HW_RNG_NRFX_API_VERSION == 2
+            // Nordic changes API between nrfx v1.7.2 and v2.1.0...
             while (!nrf_rng_event_check(NRF_RNG, NRF_RNG_EVENT_VALRDY));
             uint8_t v = nrf_rng_random_value_get(NRF_RNG);
             nrf_rng_event_clear(NRF_RNG, NRF_RNG_EVENT_VALRDY);
+#elif SIMPLEHACKS_HW_RNG_NRFX_API_VERSION == 1
+            // Nordic changes API between nrfx v1.7.2 and v2.1.0...
+            while (!nrf_rng_event_get(NRF_RNG_EVENT_VALRDY));
+            uint8_t v = nrf_rng_random_value_get();
+            nrf_rng_event_clear(NRF_RNG_EVENT_VALRDY);
+#else
+    #error "Unknown NRFX API version ... aborting compilation"
+#endif
             return v;
         }
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static void internal_fill_buffer( void * buffer, size_t count ) {

--- a/src/hw_rng.h
+++ b/src/hw_rng.h
@@ -72,9 +72,9 @@ namespace SimpleHacks {
         HW_RNG& operator=(const HW_RNG&&) =delete;
     public: // user API
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static void begin() {
-            nrf_rng_error_correction_enable();
-            nrf_rng_shorts_disable(NRF_RNG_SHORT_VALRDY_STOP_MASK);
-            nrf_rng_task_trigger(NRF_RNG_TASK_START);
+            nrf_rng_error_correction_enable(NRF_RNG);
+            nrf_rng_shorts_disable(NRF_RNG, NRF_RNG_SHORT_VALRDY_STOP_MASK);
+            nrf_rng_task_trigger(NRF_RNG, NRF_RNG_TASK_START);
         }
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static void end() {}
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static uint8_t get_uint8() {
@@ -138,9 +138,9 @@ namespace SimpleHacks {
 
     private: // to support one-time init...
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static uint8_t internal_get_byte() {
-            while (!nrf_rng_event_get(NRF_RNG_EVENT_VALRDY));
-            uint8_t v = nrf_rng_random_value_get();
-            nrf_rng_event_clear(NRF_RNG_EVENT_VALRDY);
+            while (!nrf_rng_event_check(NRF_RNG, NRF_RNG_EVENT_VALRDY));
+            uint8_t v = nrf_rng_random_value_get(NRF_RNG);
+            nrf_rng_event_clear(NRF_RNG, NRF_RNG_EVENT_VALRDY);
             return v;
         }
         SIMPLEHACKS_HW_RNG_INLINE_ATTRIBUTE static void internal_fill_buffer( void * buffer, size_t count ) {

